### PR TITLE
feat: Add Redox OS

### PR DESCRIPTION
**Pull request by zen0bit**
Need more tweaking! (configuration)

- [x] dowload - only latest version (manually set)
- [x] unpack
- [x] create VM - config!
- [X] boot VM - PARTIALLY (Need more investigation)
- [ ] configuration

Booted but! Try yourself and help...

Eventually closes: #1076

### CODE REVIEW

**Comment by lj3954** [comment]
I worked on this for a bit earlier as well, didn't realize there was newer images than the 0.8.0. Either way, we can't be hardcoding the date like that. Look at the get_ubuntu function for an example of what you should be doing. curl the SHA256SUM page, grep for the edition, grep for '.iso', and then gather the filename/checksum.

File: quickget [hunk 1]
```diff
@@ -2633,6 +2644,27 @@ function get_reactos() {
...
+    local EDITION="${1:-}"
+    local HASH=""
+    local ISO=""
+    local URL=""
+    URL="https://static.redox-os.org/img/x86_64"
+    # For now number is set manually, will see how often will be updated..
+    NUMBER="2024-04-18_975"
+    ISO="redox_${EDITION}_x86_64_${NUMBER}_livedisk.iso.zst"
+    HASH=$(wget -qO- https://static.redox-os.org/img/x86_64/SHA256SUM | grep ${ISO} | cut_1)
+    echo "${URL}/${ISO} ${HASH}"
```
Comment (lj3954) [inline quickget:2665]:
```suggestion
    local HASH=""
    local ISO=""
    local URL="https://static.redox-os.org/img/x86_64"
    local DATA

    DATA="$(curl -s "${URL}/SHA256SUM" | grep ".iso" | grep "${EDITION}")"
    ISO="$(echo ${DATA}" | cut -d '*' -f2)"
    HASH="$(echo "${DATA}" | cut_1

    echo "${URL}/${ISO} ${HASH}"
```

**Review by lj3954 — CHANGES_REQUESTED** [review]

File: quickget (hunk 1 above)
Comment (zen0bit) [reply quickget:2665]: we need use this parametrs:
```
SDL_VIDEO_X11_DGAMOUSE=0 qemu-system-x86_64 -d cpu_reset,guest_errors -smp 4 -m 2048 \
    -chardev stdio,id=debug,signal=off,mux=on,"" -serial chardev:debug -mon chardev=debug \
    -machine q35 -device ich9-intel-hda -device hda-duplex -netdev user,id=net0 \
    -device e1000,netdev=net0 -device nec-usb-xhci,id=xhci -enable-kvm -cpu host \
    -drive file=`echo $HOME/Downloads/redox_demo_x86_64*_harddrive.img`,format=raw
```

Cn you help mod quickemu...

**Comment by zen0bit** [comment]
> I worked on this for a bit earlier as well, didn't realize there was newer images than the 0.8.0. Either way, we can't be hardcoding the date like that. Look at the get_ubuntu function for an example of what you should be doing. curl the SHA256SUM page, grep for the edition, grep for '.iso', and then gather the filename/checksum.

Me to Its why I puted there just this one release...

**Comment by zen0bit** [comment]
we need use this parametrs:
```
SDL_VIDEO_X11_DGAMOUSE=0 qemu-system-x86_64 -d cpu_reset,guest_errors -smp 4 -m 2048 \
    -chardev stdio,id=debug,signal=off,mux=on,"" -serial chardev:debug -mon chardev=debug \
    -machine q35 -device ich9-intel-hda -device hda-duplex -netdev user,id=net0 \
    -device e1000,netdev=net0 -device nec-usb-xhci,id=xhci -enable-kvm -cpu host \
    -drive file=`echo $HOME/Downloads/redox_demo_x86_64*_harddrive.img`,format=raw
```

Can you help mod quickemu @lj3954

**Comment by lj3954** [comment]
> we need use this parametrs:
> 
> ```
> SDL_VIDEO_X11_DGAMOUSE=0 qemu-system-x86_64 -d cpu_reset,guest_errors -smp 4 -m 2048 \
>     -chardev stdio,id=debug,signal=off,mux=on,"" -serial chardev:debug -mon chardev=debug \
>     -machine q35 -device ich9-intel-hda -device hda-duplex -netdev user,id=net0 \
>     -device e1000,netdev=net0 -device nec-usb-xhci,id=xhci -enable-kvm -cpu host \
>     -drive file=`echo $HOME/Downloads/redox_demo_x86_64*_harddrive.img`,format=raw
> ```
> 
> Can you help mod quickemu @lj3954

I'm not interested on making major changes to bash quickemu. I'm working on my rewrite, as you already know, and I'll add support for these operating systems to that, once I get basic functionality.

**Comment by zen0bit** [comment]
> > we need use this parametrs:
> > ```
> > SDL_VIDEO_X11_DGAMOUSE=0 qemu-system-x86_64 -d cpu_reset,guest_errors -smp 4 -m 2048 \
> >     -chardev stdio,id=debug,signal=off,mux=on,"" -serial chardev:debug -mon chardev=debug \
> >     -machine q35 -device ich9-intel-hda -device hda-duplex -netdev user,id=net0 \
> >     -device e1000,netdev=net0 -device nec-usb-xhci,id=xhci -enable-kvm -cpu host \
> >     -drive file=`echo $HOME/Downloads/redox_demo_x86_64*_harddrive.img`,format=raw
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Can you help mod quickemu @lj3954
> 
> I'm not interested on making major changes to bash quickemu. I'm working on my rewrite, as you already know, and I'll add support for these operating systems to that, once I get basic functionality.

oK

**Comment by flexiondotorg** [comment]
Pause work on this until I get the `web_pipe()` branch merged :+1: 

- #1087

**Pull request closed without merging**